GlucoseKit/GlucoseStore.swift

```diff
@@ -124,7 +124,7 @@ public final class GlucoseStore: HealthKitSampleStore {
                     let sortedGlucose = glucose.sorted { $0.startDate < $1.startDate }
 
                     self.sampleDataCache.append(contentsOf: sortedGlucose)
-                    self.purgeOldGlucoseSamples()
+                    //self.purgeOldGlucoseSamples()
```

Comment:
For G4/G5 cases (non enlight), we will still want to purge old data. The comment about source preference doesn't necessarily apply to glucose. It's intended for exercise data.  Any other app that queries glucose data will get duplicates.  Dexcom is shipping backfill-enabled transmitters soon, so it's best to let its own app put data in HealthKit. 

There are (at least) two possible routes for dealing with this:

1) Set managedDataInterval public ivar to null in the case of enlite, which will prevent purging.  If we do this, the recentSamplesPredicate method should be changed make startTime non-optional (the only call to it already supplies startTime).

2) Add another metadataKey to denote long/short term storage, and modify the addGlucose* methods to accept a flag.  The different places we add data (enlite, g4, g5) would set the flag.  And the purge routine would only purge entries with the flag set.